components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java

```diff
@@ -180,6 +182,45 @@ Get ancestor organization ids (including itself) of a given organization. The li
     }
 
     @Override
+    public void clearBrandingResolverCacheHierarchy(String currentTenantDomain) throws BrandingPreferenceMgtException {
+        String organizationId = getOrganizationId();
+
+        try {
+            OrganizationManager organizationManager =
+                        BrandingResolverComponentDataHolder.getInstance().getOrganizationManager();
+
+            if (organizationId == null) {
+                // If organization id is not available in the context, try to resolve it from tenant domain
+                organizationId = organizationManager.resolveOrganizationId(currentTenantDomain);
+            }
+
+            List<String> childOrganizationIds = new ArrayList<>();
+            childOrganizationIds.add(organizationId);
```

Comment:
If we add the `organizationId` to the list, we will try to get the children again which will try to clean the cache twice, right?

Reply:
Since the initial `childOrganizationIds` list is empty, `organizationId` will be the first element to be added. Therefore, inside the loop, its children will be added to the list for the fist time as well. Thus, it will not try to clean the cache twice.